engine/src/main/resources/assets/shaders/ssao_frag.glsl

```diff
@@ -68,7 +68,15 @@ void main() {
         // TODO: Holy... frustum ray and linearized depth - please!
         sampleDepth = reconstructViewPos(texture2D(texDepth, offset.xy).r * 2.0 - 1.0, gl_TexCoord[0].xy, invProjMatrix).z;
 
-        rangeCheck = smoothstep(0.0, 1.0, ssaoRad / abs(viewSpacePos.z - sampleDepth));
+        const float maxDepthDifference = 1;
+        float depthDifference = abs(viewSpacePos.z - sampleDepth);
+
+        float rangeCheck;
+        if (depthDifference > maxDepthDifference) {
+        	rangeCheck = 0;
+        } else {
+        	rangeCheck = smoothstep(0.0, 1.0, ssaoRad / abs(viewSpacePos.z - sampleDepth));
```

Comment:
ssaoRad -> ssaoRadius

Also, after the / you can just write depthDifference, which you already calculated.

Reply:
The main reason for calculating depthDifference was so that I could re-use it here.. and yet silly me decided to calculate it twice nonetheless :D